### PR TITLE
Mq19

### PR DESCRIPTION
**Pull request by sveinbjornt**
* Added speech synthesis module for the AzureSpeech API's new Icelandic voices, Gudrun and Gunnar
* Scraper roots that are not actively being scraped are now marked as such on /stats page
* Bug fixes in currency, geography qmodules
* Typing and bug fixes in new play qmodule
* Some minor additions in special qmodule
* Fixed nasty voice pronunciation bug in words qmodule
* Web frontend for query processor now supports open_url param (URL opened in new window via target _blank)
* Some minor fixes and improvements in test_queries
* Added vb.is scraper and requisite assets

NB: All the changes in the JS files are almost entirely due to formatting imposed on them by the VSCode JS & Jinja2 plugins.

### CODE REVIEW

File: speech/voices/azure.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    text: str,
+    text_format: str,
+    audio_format: str,
+    voice_id: str,
+    speed: float,
+) -> Optional[bytes]:
+    """Returns audio data for speech-synthesised text."""
+    # Pass all arguments on to another function
+    return _azure_synthesized_text_data(**locals())
+
```
Comment (vthorsteinsson) [inline speech/voices/azure.py:150]: Why not just `text_to_audio_data = _azure_synthesized_text_data`?

File: speech/voices/azure.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+def text_to_audio_url(
+    text: str,
+    text_format: str,
+    audio_format: str,
+    voice_id: str,
+    speed: float,
+) -> Optional[str]:
+    """Returns URL to audio of speech-synthesised text."""
+    # Pass all arguments on to another function
+    return _azure_synthesized_text_url(**locals())
```
Comment (vthorsteinsson) [inline speech/voices/azure.py:161]: Ditto here

File: scrapers/default.py [hunk 3]
```diff
@@ -2035,12 +2034,35 @@ def get_metadata(self, soup):
...
+        if date_span:
+            date = ScrapeHelper.tag_prop_val(soup, "span", "class", "date")
+            datestr = date.get_text().strip()
+            # Parse date
+            try:
+                # Example: "17. janúar 2019 14:30"
+                (mday, m, y, hm) = datestr.split()
+                (hour, mins) = hm.split(":")
+                mday = mday.replace(".", "")
+                month = MONTHS.index(m) + 1
```
[the remark this answers is not in the document]
Comment (sveinbjornt) [reply scrapers/default.py:2054]: That's an unlikely edge case, and in any case, it's wrapped in a try clause. If parsing the date fails, the article gets the current date instead and scraping still succeeds.

File: speech/voices/azure.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        synthesizer = speechsdk.SpeechSynthesizer(speech_config=speech_config)
+        result = synthesizer.speak_text(text)
+
+        # Read audio from stream into buffer, append to bytearray containing total audio data
+        stream = speechsdk.AudioDataStream(result)
+        audio_data = bytearray()
+        audio_buffer = bytes(32000)  # 32 KB buffer
+        filled_size = stream.read_data(audio_buffer)
+        while filled_size > 0:
+            audio_data += audio_buffer[:filled_size]
```
[the remark this answers is not in the document]
Comment (sveinbjornt) [reply speech/voices/azure.py:112]: It would?

**Comment by vthorsteinsson** [comment]
Generally LGTM.

**Pull request merged**